### ISSUE TITLE
Cannot Build Anything 3.14 w/ CSD v14 no DLL

### ISSUE THREAD

**Issue by ghost** [Thalassicus/cep-bnw#161]
I can build with just 3.14 but if I use the non DLL version of CSD i cannot.  I had all the option.sql options set correctly.

From lua.log:

```
[335201.789] MT_Initialize: MT_Initialize.lua
[335202.058] StoryPopup:  30 ms loading Story_Events.lua
[335202.058] StoryPopup:   0 ms loading StoryPopup.lua
[335202.076] TW_Init: Initializing TW_Init.lua
[335202.389] TW_Init: \Users\WarEagleUser\Documents\Aspyr\Sid Meier's Civilization 5\MODS\Communitas AI and Tools (v 3)\Tools\TooltipWriter\TW_Init.lua:442:  Cannot find key - PercentBuildingNeeded
[335202.389] TW_Init: \Users\WarEagleUser\Documents\Aspyr\Sid Meier's Civilization 5\MODS\Communitas AI and Tools (v 3)\Tools\TooltipWriter\TW_Init.lua:445:  Cannot find key - PercentBuildingNeeded
[335202.389] Runtime Error: \Users\WarEagleUser\Documents\Aspyr\Sid Meier's Civilization 5\MODS\Communitas AI and Tools (v 3)\Tools\TooltipWriter\TW_Init.lua:549: bad argument #5 to 'ConvertTextKey' (Argument must be [boolean, number, or string].)
stack traceback:
    =[C]: in function 'ConvertTextKey'
    \Users\WarEagleUser\Documents\Aspyr\Sid Meier's Civilization 5\MODS\Communitas AI and Tools (v 3)\Tools\TooltipWriter\TW_Init.lua:549: in function 'InsertBuildingSubStat'
    \Users\WarEagleUser\Documents\Aspyr\Sid Meier's Civilization 5\MODS\Communitas AI and Tools (v 3)\Tools\TooltipWriter\TW_Init.lua:447: in function 'GetDefaultBuildingStatText'
    \Users\WarEagleUser\Documents\Aspyr\Sid Meier's Civilization 5\MODS\Communitas AI and Tools (v 3)\Tools\TooltipWriter\TW_Init.lua:661: in main chunk
    =[C]: ?
[335202.389] Runtime Error: Error loading \Users\WarEagleUser\Documents\Aspyr\Sid Meier's Civilization 5\MODS\Communitas AI and Tools (v 3)\Tools\TooltipWriter\TW_Init.lua.
[335202.610] CEA_Events: CEA_Events.lua
[335202.727] CEA_Promotions: CEA_Promotions.lua
[335202.837] CEB_Events: CEB_Events.lua
[335204.047] AdvisorInfoPopup: Closing Advisor Info
[335204.047] Demographics: Dequeuing demographics
[335204.047] Demographics: Dequeuing demographics
[335205.652] LoadScreen: WARN   Turn 0   GetHelpTextForBuilding: stat data is nil for BUILDING_INDUS_SANITATION!
[335205.652] LoadScreen: WARN   Turn 0   GetHelpTextForBuilding: stat data is nil for BUILDING_VEDI!
[335205.655] CEAI_Events: CheckPlayerStartBonuses
[335205.685] CEAI_Events: CheckPlayerStartBonuses Done
[335205.686] CET_Events: UpdatePlotYields Start
[335205.691] CET_Events: UpdatePlotYields Done
[335210.137] UnitFlagManager: WARN   Turn 0   Flag not found for OnNewUnitRefresh: Player[5] Unit[32771]
[335216.894] TechPopup: WARN   Turn 0   GetHelpTextForBuilding: stat data is nil for BUILDING_GRANARY!
[335216.910] TechPopup: WARN   Turn 0   GetHelpTextForBuilding: stat data is nil for BUILDING_TEMPLE_ARTEMIS!
[335216.920] TechPopup: WARN   Turn 0   GetHelpTextForBuilding: stat data is nil for BUILDING_HEROIC_EPIC!
[335217.923] TechPopup: stealingTechTargetPlayerID: -1
[335217.924] TechPopup: iValue: 0
[335218.984] ProductionPopup: WARN   Turn 0   GetHelpTextForPromotion: stat data is nil for PROMOTION_IGNORE_TERRAIN_COST_NOUPGRADE!
[335218.984] Runtime Error: \Users\WarEagleUser\Documents\Aspyr\Sid Meier's Civilization 5\MODS\Communitas AI and Tools (v 3)\Tools\TooltipWriter\InfoTooltipInclude.lua:166: attempt to index local 'section' (a nil value)
```

**Comment by ghost** [comment]
Also I am on a Mac so that is why I am using the non DLL version.

**Comment by skodkim** [comment]
I think you have set using_cep to 1 in csd's option file, right?

This is a remnant from God's and kings in the csd file so try changing it so you say you're not using CEP.

\Skodkim?

Jerrod Moll notifications@github.com skrev:

> I can build with just 3.14 but if I use the non DLL version of CSD i
> cannot.  I had all the option.sql options set correctly.
> 
> From lua.log:
> 
> [334301.904] TW_Init: Initializing TW_Init.lua
> [334302.239] TW_Init: \Users\WarEagleUser\Documents\Aspyr\Sid Meier's
> Civilization 5\MODS\Communitas AI and Tools (v
> 3)\Tools\TooltipWriter\TW_Init.lua:442:  Cannot find key -
> PercentBuildingNeeded
> [334302.239] TW_Init: \Users\WarEagleUser\Documents\Aspyr\Sid Meier's
> Civilization 5\MODS\Communitas AI and Tools (v
> 3)\Tools\TooltipWriter\TW_Init.lua:445:  Cannot find key -
> PercentBuildingNeeded
> [334302.239] Runtime Error: \Users\WarEagleUser\Documents\Aspyr\Sid
> Meier's Civilization 5\MODS\Communitas AI and Tools (v
> 3)\Tools\TooltipWriter\TW_Init.lua:549: bad argument #5 to
> 'ConvertTextKey' (Argument must be [boolean, number, or string].)
> stack traceback:
>   =[C]: in function 'ConvertTextKey'
>   \Users\WarEagleUser\Documents\Aspyr\Sid Meier's Civilization
> 5\MODS\Communitas AI and Tools (v
> 3)\Tools\TooltipWriter\TW_Init.lua:549: in function
> 'InsertBuildingSubStat'
>   \Users\WarEagleUser\Documents\Aspyr\Sid Meier's Civilization
> 5\MODS\Communitas AI and Tools (v
> 3)\Tools\TooltipWriter\TW_Init.lua:447: in function
> 'GetDefaultBuildingStatText'
>   \Users\WarEagleUser\Documents\Aspyr\Sid Meier's Civilization
> 5\MODS\Communitas AI and Tools (v
> 3)\Tools\TooltipWriter\TW_Init.lua:661: in main chunk
>   =[C]: ?
> [334302.239] Runtime Error: Error loading
> \Users\WarEagleUser\Documents\Aspyr\Sid Meier's Civilization
> 5\MODS\Communitas AI and Tools (v 3)\Tools\TooltipWriter\TW_Init.lua.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/Thalassicus/cep-bnw/issues/161

## 

Sendt fra min Android telefon med K-9 Mail. Undskyld hvis jeg er lidt kortfattet.

**Comment by ghost** [comment]
That was it!  just need to set options in CAT and not CSD!  Been awhile since I have played with CSD :)

**Comment by skodkim** [comment]
Good!

I've asked Gazebo to remove the option from CSD.

\Skodkim

**Comment by stackpoint** [comment]
CSD v14 should be (non-crashing) compatible with CSD even without USING_CSD set to 1 because it doesn't use lua files. This might be a different problem.

**Comment by stackpoint** [comment]
The RequiresBuildingInPercentCities error is interesting since it should never trigger since CEP and CSD don't have any references to it....

**Comment by ghost** [comment]
i also had city limit, calypso's colored religion and robe's global relations mods enabled.  once i changed the reference in the CSD option.sql file back everything worked find for me.

**Comment by skodkim** [comment]
The problem is due to a setting in the csd options file. I think it's called using_cep. If enabled it makes national wonders buildable with the required buildings in only 75% of your cities. This is however a leftover from gem and its not used in cep. I have written it in a thread at the comminitas forums and have made gazebo remove the option for future releases.

I've also discussed potential problems with some of his other settings (optional tweaks to policies, wonder and Sweden) with him but even if these settings are enabled they should not result in anything fatal, just unintended behavior.

\Skodkim

Jerrod Moll notifications@github.com skrev:

> Closed #161.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/Thalassicus/cep-bnw/issues/161

## 

Sendt fra min Android telefon med K-9 Mail. Undskyld hvis jeg er lidt kortfattet.
